### ISSUE TITLE
Can't get FiltaQuilla to change a tag when receive messages that have a certain tag

### ISSUE THREAD

**Issue by bkarlan** [RealRaven2000/FiltaQuilla#213]
I have a tag, `@waiting` for emails where I'm waiting for a response.  I want FiltaQuilla to change the status of any incoming email that has this tag to `@followup`.  I have your add on installed and I have a Message Filter for each and every email address I have, 10 of them, to change the tag from `@waiting` to `@followup` when receiving emails with `@waiting`.  Yet, it doesn't work.  I downloaded your add on specifically for this purpose and no matter how I configure it, I cannot get it to work upon email receipt.  It will work if I run it manually, but then it changes everything, including all the emails I'm waiting for a response on.

**Comment by RealRaven2000** [comment]
Does the filter have additional search terms (apart from the tag being `waiting`)

Can you export the filters with quickFilters Pro? It has a feature to export a subset of filter (instead of all of them) if you hold down CTRL while clicking on backup:

![image](https://user-images.githubusercontent.com/6314674/220178727-1b8c7fa1-80d8-41cf-b5b2-ef6a5af9a74e.png)

**Comment by RealRaven2000** [comment]
There is also a troubleshooter in case the filter was merged together and has mixed operators:

![image](https://user-images.githubusercontent.com/6314674/220178822-2c856b70-5b67-4e18-9eee-577d33bddbcb.png)

**Comment by bkarlan** [comment]
Not sure what you are looking at as it doesn't look anything like what I have.  I have no such toolbar.![image](https://user-images.githubusercontent.com/2250480/220188846-8b3abc62-2385-473a-8781-0bbac945409d.png)![image](https://user-images.githubusercontent.com/2250480/220188866-5d41cff2-e31d-415b-a555-c1ab4aafa571.png)

**Comment by RealRaven2000** [comment]
That toolbar is provided by my Add-on quickFilters.

**Comment by RealRaven2000** [comment]
That toolbar is provided by my [Add-on quickFilters](https://addons.thunderbird.net/thunderbird/addon/quickfilters/?src=ss). It finances all the work done here in FiltaQuilla too.

**Comment by bkarlan** [comment]
Ok, so I installed this extra add on.  But what does it do for me?  I tried the troubleshooter but it found no problems.

**Comment by RealRaven2000** [comment]
having no broken filters is already a result.

as next step, you could use the backup feature to export the filter in
question so I can import it and have a look / test it here on my system.

select the filter, then hold CTRL while clicking backup. Also, what process
sets the first tag?

**Comment by bkarlan** [comment]
I did make a copy of it above.  The 2nd pic I shared is the filter.  It's pretty basic.

**Comment by RealRaven2000** [comment]
who / what process sets the flag to @waiting in the first place? if it's a
timing issue it may help to activate the option " run filter periodically ".

**Comment by bkarlan** [comment]
Getting new mail, after junk classification.  I've tried it before junk classification, but it doesn't work either.

**Comment by RealRaven2000** [comment]
> Getting new mail, after junk classification. I've tried it before junk classification, but it doesn't work either.

Let me ask the question again. WHAT sets the tag to @waiting ???? (before the filter can react)

**Comment by bkarlan** [comment]
I set it manually to @waiting.  This is specifically for emails that I want to track where I’m waiting for a response.

**Comment by RealRaven2000** [comment]
> I set it manually to @waiting. This is specifically for emails that I want to track where I’m waiting for a response.

Thanks - that was the missing bit of information. So now I can test this stuff.

**Comment by RealRaven2000** [comment]
> I set it manually to @waiting. This is specifically for emails that I want to track where I’m waiting for a response.

Just reading that again, but you also wrote:

> I have a tag, @waiting for emails where I'm waiting for a response. I want FiltaQuilla to change the status of any incoming email that has this tag to @followup. 

So... you set it @waiting manually when it is in your inbox. So how can it be an incoming that has this tag? I really don't understand it. It would have to already have the tag when Thunderbird receives it, right? So you're not setting it to @waiting manually?

**Comment by bkarlan** [comment]
Perhaps I didn't explain it well.  I have an email in my sent folder that I am waiting for a response on.  I tag it to `@waiting`.  I want the system to change this tag to `@followup` once a reply is received on that email that has that tag.

**Comment by RealRaven2000** [comment]
> Perhaps I didn't explain it well. I have an email in my sent folder that I am waiting for a response on. I tag it to `@waiting`. I want the system to change this tag to `@followup` once a reply is received on that email that has that tag.

Right. Filters cannot run automatically on the Sent mail folder. Only in the Inbox. This is by design (Thunderbird).

**Comment by bkarlan** [comment]
I think you are on to something.  I was aware that it can't run on the Sent folder and that's not what I was asking it to do.  However, I think what is happening is that the tag isn't carrying over into the response to the email that is flagged.  If it were, then the filter would be triggered.  But the tag is only on the original email in the sent folder.  So is there a way to have the flag carry over into the associated response?

**Comment by RealRaven2000** [comment]
you're right: replies do not inherit tags from previous emails. 
I don't think this is possible with filters. but it is an interesting use case; it may be a possible new feature for my Add-on SmartTemplates Pro. this already has a lot of functionality and syntax to read and modify mail headers. since it runs automatically when you reply to an email, it is also aware of the original email. 

There is a syntax `%header.set(hdr,value)%` that can be used to write new subject lines or add recipients. it could be extended to add tags. likewise, a new variable %tags% could be implemented to read the tag(s) from original email. the commands in the mail template could then look something like this:

```
%tags(toclipboard)% 
%header.set(tags,clipboard)%
```

The reason I would use clipboard to transfer this data from the original header to the new email is that the functions cannot be "nested".(so you cannot write `%header.set(tags,%tags%)%` )

**Comment by RealRaven2000** [comment]
hmm reading this again... this approach only works if you reply to the email (you can then read the tags of the original mail, and modify your reply). what you need is the opposite: the original email to be modified. and that would need to be based on references in the reply email. at the moment I cannot think of a clever way to do this...

**Comment by bkarlan** [comment]
Smallcubed has a Mac Mail add on called Mail Suite and it used to allow this to occur.  In fact they had a specific setting for it.  I don’t use Mail anymore so I don’t know if they still have this functionality.

**Comment by RealRaven2000** [comment]
> Perhaps I didn't explain it well. I have an email in my sent folder that I am waiting for a response on. I tag it to `@waiting`. I want the system to change this tag to `@followup` once a reply is received on that email that has that tag.

Right. Filters cannot run automatically on the Sent mail folder. Only in the Inbox. This is by design (Thunderbird).

**Comment by ThiloteE** [comment]
### Ugly complex workaround using filters and tags:

1. Create filter (1) that moves or copies all sent mails From Inbox A to an inbox B and tags them with `X`.
      An inbox  can be:
      
      - The inbox of your e-mail account
      - The inbox of a separate e-mail account
      - The inbox of a feed
      - The inbox of your local folder

   You will have to execute this filter manually via `Tools > Run filters on folder`

2. Create filter (2) for Inbox A with filter condition "contains `Y`" and filter action that copies, sends or forwards these incoming mails to Inbox B. This step here is the point where you notice somebody has replied to your sent mail. You will have to execute this filter manually via `Tools > Run filters on folder` or alternatively let the filter run automatically every 10 minutes. Unfortunately it is not possible to adjust the 10 minutes to a different time frame.

3. Create filter (3) for Inbox B with filter condition contains `Y` and filter action: remove tag `X`.

### Comment:

Not something I would recommend for efficient workflow.
In particular, your filter conditions will likely trigger on mails you never intended them to be triggered on.

### Ugly workaround using "Tasks" and "Search":


1. Whenever you send a mail, mention an `issue number` in subject or body of the mail. Ask your recipients when they respond to mention the issue number.
2. convert this sent mail to a task. (When you open the task the sent mail is linked, so you will be able to find the sent mail.)
3. Whenever you receive a mail, search through your task list for the issue number.

obviously you can also use this with tags, but I think the tasks feature is actually quite capable. It would be neat, if filters, tags, tasks and events woudl be linked more closely, but alas, this is what we have right now...
